content/guide/core/file-system.md

```diff
@@ -0,0 +1,747 @@
+### isKnown
+
+```ts
+isItAKnownFolder: boolean = folder.isKnown
+```
+Determines whether this instance is a known folder (accessed through the `knownFolders` object).
+
+---
+Both the File and Folder classes extend the FileSystemEntity which has the following API:
+
+### lastModified
+
+```ts
+lastModified: Date = entity.lastModified 
+```
+
+Gets the Date object specifying the last time this entity was modified.
+
+---
+
+### name
+
+```ts
+name: string = entity.name
+```
+
+Gets the name of the entity.
+
+---
+
+### path 
+
+```ts
+path: string = entity.path
+```
+
+Gets the fully-qualified path (including the extension for a File) of the entity.
+
+---
+
+### parent
+
+```ts
+parent : Folder = entity.parent
+```
+Gets the Folder object representing the parent of this entity.
+Will be null for a root folder like Documents or Temporary.
+This property is readonly.
+	
+---
+
+### remove()
+
+```ts
+result: any = await entity.remove()
+```
+
+Asynchronously removes (deletes) the current Entity from the file system.
+
+---
+
+### removeSync()
+
+```ts
+entity.removeSync(onError?: (error: any) => any): void
+```
+
+Removes (deletes) the current Entity from the file system.
+
+---
+
+### rename()
+
+```ts
+entity.rename(newName: string): Promise<any>
+```
+
+Asynchronously renames the current entity using the specified name.
+
+---
+
+### renameSync()
+
+```ts
+entity.renameSync(newName: string, onError?: (error: any) => any)
+```
+
+Renames the current entity synchronously, using the specified name.
+
+---
+
+### getEntities()
+
+```ts
+folder.getEntities(): Promise<Array<FileSystemEntity>>
+```
+Asynchronously gets all the top-level entities residing within a folder.
+
+---
+
+### getEntitiesSync()
+
+```ts
+folder.getEntitiesSync(onError?: (error: any) => any): Array<FileSystemEntity>
+```
+
+Gets all the top-level entities residing within this folder synchronously. `onError` is a optional function to be called if some error occurs.
+
+---
+
+### eachEntity()
+
+```ts
+folder.eachEntity(onEntity: (entity: FileSystemEntity) => boolean)
+```
+
+Enumerates all the top-level FileSystem entities within a folder. `onEntity` is a callback that receives the current entity.
+
+---
+### getFile()
+```ts
+folder.getFile(name: string): File
+```
+Gets or creates a File entity with the specified name within this Folder
+
+---
+
+### extension
+
+```ts
+fileExt: string = file.extension
+```
+
+Gets the extension of the file.
+
+---
+
+### size
+
+```ts
+fileSize: number = file.size
+```
+
+Gets the extension of the file.
+
+---
+
+### isLocked
+
+```ts
+isFileLocked: boolean = file.isLocked
+```
+
+Gets a boolean value indicating whether the file is currently locked, meaning a background operation associated with this file is running.
+
+---
+
+### readText()
+
+```ts
+text: string = await file.readText(encoding)
+```
+
+Asynchronously reads the content of the file as a string using an optional encoding value. If you do not pass any encoding, `UTF-8` is used.
```

Comment:
Asynchronously reads the content of the file as a string using an optional encoding value. If you do not pass any encoding, `UTF-8` is used by default.